tools/apps/src/lib.rs

```diff
@@ -918,18 +945,17 @@ impl HttpConn for Http3Conn {
                     conn.stream_shutdown(stream_id, quiche::Shutdown::Read, 0)
                         .unwrap();
 
-                    let (headers, body) =
+                    let (headers, body, priority) =
                         Http3Conn::build_h3_response(root, index, &list);
 
-                    match self
-                        .h3_conn
-                        .send_response(conn, stream_id, &headers, false)
-                    {
+                    match self.h3_conn.send_response_with_priority(
+                        conn, stream_id, &headers, &priority, false,
+                    ) {
                         Ok(v) => v,
 
                         Err(quiche::h3::Error::StreamBlocked) => {
                             let response = PartialResponse {
-                                headers: Some(headers),
+                                headers: Some(headers.to_vec()),
```

Comment:
No need to copy this AFAICT.
```suggestion
                                headers: Some(headers),
```